src/mouse/events.rs

```diff
@@ -0,0 +1,186 @@
+//! module for abstracting over most mouse events.
+//! Handles most common mouse events so that widgets don't have to
+//! store any mouse state.
+
+use input::MouseButton;
+use position::Point;
+use time::SteadyTime;
+use ::vecmath::vec2_sub;
+
+#[cfg(test)]
+use position::Scalar;
+
+// pub type MouseEventIterator = ();
+
+/// Used for simplified mouse event handling. Most widgets can probably
+/// just use these events
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum MouseEvent {
+    /// When the mouse button has been pressed but not yet released or dragged.
+    /// This event is created as soon as the button is pressed.
+    Down(ButtonDownEvent),
+    /// Indicates that the mouse was clicked. A Click event is created when the mouse button is released, not depressed
+    Click(MouseClick),
+    /// Drag event is created when the mouse was moved over a certain threshold while a button was depressed
+    Drag(MouseDragEvent),
+    /// Scroll event is created when whenever the scroll wheel is moved
+    Scroll(Scroll),
+}
+
+
+/// Info on a simple mouse click event. This event gets dispatched when a
+/// mouse button goes down then up without moving more than the drag threshold
+/// while the button is depressed.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct MouseClick {
```

Comment:
We are already in the `mouse::event` module, so we could probably change `MouseClick` to just `Click`. This way accessing it via the module would be a little cleaner, i.e. `mouse::event::Click`. We can still re-export it from the lib.rs as `MouseClickEvent`, i.e. `use mouse::event::Click as MouseClickEvent` if we deem it necessary.